### PR TITLE
Make sure we export kubecfg before trying to dump nodes

### PR DESCRIPTION
**Pull request by justinsb**
Otherwise we might be dumping the wrong cluster

### CODE REVIEW

**Comment by justinsb** [comment]
Builds on #5244

File: kubetest/dump.go [hunk 1]
```diff
@@ -76,51 +76,99 @@ func newLogDumper(sshClientFactory sshClientFactory, artifactsDir string) (*logD
...
+	log.Printf("Dumping node %s", name)
+
+	n, err := d.connectToNode(ctx, name, ip)
+	if err != nil {
+		return fmt.Errorf("could not connect: %v", err)
+	}
+
+	errors := n.dump(ctx)
+	for _, e := range errors {
+		log.Printf("error dumping node %s: %v", name, e)
```
Comment (krzyzacy) [inline kubetest/dump.go]: so for now failing to dump logs is considered 'fine'?

File: kubetest/dump_test.go [hunk 2]
```diff
@@ -376,3 +376,71 @@ func (m *mockSSHClient) ExecPiped(ctx context.Context, command string, stdout io
...
+			expected: []string{"10.0.3.3"},
+		},
+	}
+
+	for _, g := range grid {
+		actual := findInstancesNotDumped(g.ips, g.dumped)
+
+		if !reflect.DeepEqual(actual, g.expected) {
+			t.Errorf("unexpected result from findInstancesNotDumped.  actual=%v, expected=%v", actual, g.expected)
+			continue
```
Comment (krzyzacy) [inline kubetest/dump_test.go]: nit: don't need this continue?

File: kubetest/kops.go [hunk 3]
```diff
@@ -357,18 +358,64 @@ func (k kops) DumpClusterLogs(localPath, gcsPath string) error {
...
-		return fmt.Errorf("Failure exporting kops kubecfg: %v", err)
+		return fmt.Errorf("failure from 'kops export kubecfg %s': %v", k.cluster, err)
 	}
+
+	// Double-check that the file was exported
+	info, err = os.Stat(k.kubecfg)
+	if err != nil {
+		return fmt.Errorf("kubeconfig file %s was not exported", k.kubecfg)
+	}
+	if info.Size() > 0 {
```
Comment (krzyzacy) [inline kubetest/kops.go]: `info.Size() == 0`?

File: kubetest/kops.go [hunk 4]
```diff
@@ -357,18 +358,64 @@ func (k kops) DumpClusterLogs(localPath, gcsPath string) error {
...
+		return fmt.Errorf("exported kubeconfig file %s was empty", k.kubecfg)
+
```
Comment (krzyzacy) [inline kubetest/kops.go]: nit: extra empty line

**Comment by krzyzacy** [comment]
/assign
lgtm besides the stat check, and a few bikesheds

File: kubetest/dump.go (hunk 1 above)
Comment (justinsb) [reply kubetest/dump.go]: I believe that's the behaviour of log-dump.sh as well e.g. https://github.com/kubernetes/kubernetes/blob/master/cluster/log-dump/log-dump.sh#L156-L157

I added a comment along with a TODO that a cleanup is in order.

**Comment by justinsb** [comment]
Thanks for the review @krzyzacy ... made the suggested changes (I believe).

**Comment by krzyzacy** [comment]
(travis)

**Comment by krzyzacy** [comment]
(screw travis)

**Pull request merged**